tests/tests.js

```diff
@@ -519,9 +519,228 @@ describe('i18n tests', function(){
+                { code: "*", quality: 1.0 }
+            ]);
 
         result.Name.should.eql('pip pip tally ho crumpets and tea');
     });
 });
+
+describe('i18n tansformByFields tests', function () {
+
+    describe('when all requested locales are present', function () {
+        var translations = null;
+        beforeEach(function () {
+            translations = {
+                i18n: [
+                    {
+                        Name: "gonna drink some beer and shoot some stuff y'all",
+                        Description: "This is a description in US English",
+                        Language: {
+                            IETF: "en-US",
+                            Code: "en",
+                            Region: "US"
+                        }
+                    },
+                    {
+                        Name: "pip pip tally ho crumpets and tea",
+                        Description: "This is a description in GB English",
+                        Language: {
+                            IETF: "en-GB",
+                            Code: "en",
+                            Region: "GB"
+                        }
+                    }
+                ],
+                PrimaryLanguage: "en-US"
+            };
+        });
+
+        it('should return all required fields in the requested locale', function () {
+
+            var result = i18n.transformByField(translations, [{
+                code: "en",
+                region: "US",
+                quality: 1.0
+            }], {
+                required: [
+                    'Name',
+                    'Description'
+                ],
+                optional: []
+            });
+
+            result.translations.Name.should.eql("gonna drink some beer and shoot some stuff y'all");
+            result.translations.Description.should.eql("This is a description in US English");
+            result.localization.Name.should.eql('en-US');
+            result.localization.Description.should.eql('en-US');
+        });
+
+        it('should fallback to primary locale when * is used', function () {
+            var result = i18n.transformByField(translations, [{
+                code: "*",
+            }], {
+                required: [
+                    'Name',
+                    'Description'
+                ],
+                optional: []
+            });
+
+            result.translations.Name.should.eql('gonna drink some beer and shoot some stuff y\'all');
+            result.translations.Description.should.eql('This is a description in US English');
+
+            result.localization.Name.should.eql('en-US');
+            result.localization.Description.should.eql('en-US');
+
+        });
+
+        it('should fallback to primary locale when * is anywhere in the set', function () {
+            var result = i18n.transformByField(translations, [
+                { code: "es", region: "MX", quality: 1.0 },
+                { code: "*", quality: 0.8 },
```

Comment:
I wonder if this case is realistic. If you put a wildcard with higher quality than a language, is that language relevant in the list? I guess it's just for completeness.